developer_guides/apps/kd_integration/images/kd-e2e-sequence-diagram.pu

```diff
@@ -0,0 +1,110 @@
+
+participant "Userspace component" as usr
+participant "Audio driver" as drv
+participant "FW infrastructure" as fw
+participant "Data transfer to Host" as dma
+participant "Keyword detection algorithm" as kda
+participant "Data transfer to DSP" as gpdma
+
+box "Linux User/Kernel space" #LavenderBlush
+	participant usr
+	participant drv
+end box
+
+box "DSP" #LightBlue
+	participant fw
+	participant dma
+	participant kda
+	participant gpdma
+end box
+
+
+drv -> fw : Setup audio topology \n (Speech Capture & Keyword Detection pipes)
+usr -> drv : Prepare & Open PCM capture \n(snd_pcm_open/snd_pcm_hw_params)
+drv -> fw : Stream Open & Preparation
+drv -> fw : HW Params
+group optional (depends on keyword detection algorithm implementation)
+ usr -> drv : Send keyword detection algorithm parameters \n (snd_ctl_elem_tlv_write)
+ drv -> fw : Send keyword detection algorithm parameters
+ fw -> kda : Send keyword detection algorithm parameters
+end
+
+drv ->drv : DAPM power up event
+drv -> fw : HW Params for Keyphrase Detection Pipeline
+usr -> drv : Trigger start (alsamixer)
+drv -> fw : Keyword detection algorithm & buffer manager triggered
+fw -> gpdma 
+
+activate gpdma
+
+fw -> kda : keypharse detection enabled
+
+activate kda
+
+usr -> drv : Trigger start (snd_pcm_read)
+
+note over usr
+Speech application indefinitely 
+waits for data.
+end note 
+
+ref over usr, drv, fw , gpdma, kda, dma  
+Speech Capture pipeline is not transmitting data to Host system
+Host system may enter the low power state
+end ref
+
+loop keyword detection algorithm \nexecuted on DSP
+ kda <- gpdma 
+end
+
+hnote over kda : keyword is detected
+
+fw <-- kda : notification on keyword detection
```

Comment:
I think we agreed to call it *event* on FW side.

Reply:
good catch, I've updated the diagram to keep terminology consistent